application-ckeditor-ui/src/main/resources/CKEditor/EditSheet.xml

```diff
@@ -285,8 +287,29 @@ define('xwiki-ckeditor', [
       sourceDocument = new XWiki.Document(XWiki.Model.resolve(sourceDocumentReference, XWiki.EntityType.DOCUMENT));
     }
     sourceDocument.syntax = $(element).attr('data-sourceDocumentSyntax');
+    var isHTML5 = $(element).data('syntax') !== 'annotatedxhtml/1.0';
+
+    var allowedContent;
+    var allowedContentWithoutFigure;
+    if (sourceDocument.syntax in allowedContentBySyntax) {
+      allowedContent = allowedContentBySyntax[sourceDocument.syntax];
+      // Try removing the figure/figcaption tags for caption content to forbid nested figures.
+      allowedContentWithoutFigure = $.extend(true, {}, allowedContent);
+      if ('$1' in allowedContentWithoutFigure &amp;&amp; 'elements' in allowedContentWithoutFigure['$1']) {
+        delete allowedContentWithoutFigure['$1'].elements.figure;
+      }
```

Comment:
```suggestion
      delete allowedContentWithoutFigure['$1']?.elements?.figure;
```

Reply:
Actually this change doesn't work, the compiler that is supposed to make this compatible with older browsers introduces a bug such that the `delete` isn't executed. Reverted in 3cb2454580508edacbe60dbd97af7aff0bce136a to fix [CKEDITOR-433](https://jira.xwiki.org/browse/CKEDITOR-433).